asserts/database.go

```diff
@@ -461,3 +479,45 @@ var DefaultCheckers = []Checker{
 	CheckTimestampVsSigningKeyValidity,
 	CheckCrossConsistency,
 }
+
+// no-authority assertion checkers
+
+// NoAuthorityCheckSignature checks that the signature is valid.
+func NoAuthorityCheckSignature(assert Assertion, roDB RODatabase, checkTime time.Time) error {
+	selfSigned, ok := assert.(selfSignedAssertion)
+	if !ok {
+		panic(fmt.Errorf("cannot check non-self-signed assertion type %q", assert.Type().Name))
+	}
+	signingKey := selfSigned.signKey()
+	content, encSig := assert.Signature()
+	signature, err := decodeSignature(encSig)
+	if err != nil {
+		return err
+	}
+	err = signingKey.verify(content, signature)
+	if err != nil {
+		return fmt.Errorf("failed signature verification: %v", err)
+	}
+	return nil
+}
+
+// A noAuthorityConsistencyChecker performs further checks based on the full
+// assertion database knowledge and its own signing key.
+type noAuthorityConsistencyChecker interface {
+	noAuthorityCheckConsistency(roDB RODatabase) error
```

Comment:
Fair enough, missed that.

On Fri, 2 Sep 2016, 17:52 Colin Watson, <notifications@github.com> wrote:

> In asserts/database.go
> <https://github.com/snapcore/snapd/pull/1791#discussion_r77377613>:
>
> > +	content, encSig := assert.Signature()
> > +	signature, err := decodeSignature(encSig)
> > +	if err != nil {
> > +		return err
> > +	}
> > +	err = signingKey.verify(content, signature)
> > +	if err != nil {
> > +		return fmt.Errorf("failed signature verification: %v", err)
> > +	}
> > +	return nil
> > +}
> > +
> > +// A noAuthorityConsistencyChecker performs further checks based on the full
> > +// assertion database knowledge and its own signing key.
> > +type noAuthorityConsistencyChecker interface {
> > +	noAuthorityCheckConsistency(roDB RODatabase) error
>
> I don't think that makes sense in this case: checkConsistency (which this
> is modelled on) doesn't have it.
>
> —
> You are receiving this because you were mentioned.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/snapcore/snapd/pull/1791/files/3067c8b60100887f170ad0b2128e14b6ee43a213#r77377613>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAA60jg-l7bY6GguA-AJaYc6lCByYDFBks5qmFREgaJpZM4JwtQi>
> .
>